canvas_modules/common-canvas/src/common-properties/controls/list/list.jsx

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright 2017-2020 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from "react";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+import NumberfieldControl from "./../numberfield";
+import TextfieldControl from "./../textfield";
+import AbstractTable from "./../abstract-table.jsx";
+import FlexibleTable from "./../../components/flexible-table";
+import MoveableTableRows from "./../../components/moveable-table-rows";
+import ValidationMessage from "./../../components/validation-message";
+import { formatMessage } from "./../../util/property-utils";
+import { getDataId } from "./../../util/control-utils";
+import { TABLE_SCROLLBAR_WIDTH, MESSAGE_KEYS, STATES } from "./../../constants/constants.js";
+import { Type } from "./../../constants/form-constants.js";
+
+const NUMBER_TYPES = [Type.INTEGER, Type.DOUBLE, Type.LONG];
+
+class ListControl extends AbstractTable {
+	constructor(props) {
+		super(props);
+		this.addRow = this.addRow.bind(this);
+		this.reactIntl = props.controller.getReactIntl();
+	}
+
+	addRow() {
+		const controlPropType = this.props.controller.getControlPropType(this.props.propertyId);
+		const newRow = NUMBER_TYPES.indexOf(controlPropType) > -1 ? null : "";
+		const rows = this.props.controller.getPropertyValue(this.props.propertyId).concat([newRow]);
+		this.setCurrentControlValueSelected(rows);
+	}
+
+	makeCell(control, propertyId, controlPropType) {
+		if (NUMBER_TYPES.indexOf(controlPropType) > -1) {
+			return (<div className="properties-list-number-control">
+				<NumberfieldControl
+					control={control}
+					propertyId={propertyId}
+					controller={this.props.controller}
+					tableControl
+				/>
+			</div>);
+		}
+		return (<div className="properties-list-string-control">
+			<TextfieldControl
+				control={control}
+				propertyId={propertyId}
+				controller={this.props.controller}
+				tableControl
+			/>
+		</div>);
+	}
+
+	makeRows(controlValue, tableState) {
+		const rows = [];
+		if (controlValue) {
+			for (var rowIndex = 0; rowIndex < controlValue.length; rowIndex++) {
+				const columns = [];
+				const row = { row: rowIndex };
+				const propertyId = Object.assign({}, this.props.propertyId, row);
+				const control = Object.assign({}, this.props.control);
```

Comment:
@matthoward366  continuing the thread from https://github.com/elyra-ai/canvas/pull/171/files#r458109412
The list control is creating its own rows of TextfieldControl/NumberfieldControl passing in its own information. Unlike other complexType controls, we don't have parameter information to create the control from.

Should we use default values for the control here? In the case of textfields, something like 
```
{
  controlType: "textfield",
  editStyle: "inline",
  valueDef: {
    isList: false
    isMap: false
    propType: "string"
  }
}
```

Reply:
Can we just pass down an empty control?  Something like control={}.  It looks like the textfield has checks that default correctly.